gensim/similarities/termsim.py

```diff
@@ -293,17 +353,43 @@ def inner_product(self, X, Y, normalized=False):
         References
         ----------
         The soft cosine measure was perhaps first described by [sidorovetal14]_.
+        Further notes on the efficient implementation of the soft cosine measure are described by
+        [novotny18]_.
 
         .. [sidorovetal14] Grigori Sidorov et al., "Soft Similarity and Soft Cosine Measure: Similarity
            of Features in Vector Space Model", 2014, http://www.cys.cic.ipn.mx/ojs/index.php/CyS/article/view/2043/1921.
 
+        .. [novotny18] Vít Novotný, "Implementation Notes for the Soft Cosine Measure", 2018,
+           http://dx.doi.org/10.1145/3269206.3269317.
+
         """
         if not X or not Y:
             return self.matrix.dtype.type(0.0)
 
+        if normalized in (True, False):
+            warnings.warn(
+                'Boolean parameter normalized will be removed in 4.0.0, use '
+                'normalized=({normalized}, {normalized}) instead of '
+                'normalized={normalized}'.format(normalized=normalized),
```

Comment:
Sorry, fixed in be6810e2.

Reply:
Actually, can we discuss this? Gensim 3.8.3 supports Python 2.7 and Python 3.5+ and Gensim 4.0 only supports Python 3.5+. The `string.format` is available in Python 2.7 and since Python 3 as per [PEP 3101](https://www.python.org/dev/peps/pep-3101/), i.e. its usage should be non-problematic in both Gensim 3.8.3 and Gensim 4.0.

Reply:
It's a code style issue, not a compatibility issue. My understanding is @piskvorky dislikes .format, likes f-strings and is OK with the old-school percent formatting. This is my preference as well.

Reply:
I see. Aren't f-strings only available in Python 3.6+, whereas Gensim supports Python 3.5+, though? [[PEP 498]]

 [PEP 498]: https://www.python.org/dev/peps/pep-0498/